app/models/Event.py

```diff
@@ -265,22 +265,62 @@ def human_readable_time(self):
         :returns: The formatted date.
         :rtype: str
         """
-        output = ''
-        if self.start_time.strftime("%p") == self.end_time.strftime("%p"):
-            format = "%I:%M-"
-        else:
-            format = "%I:%M%p-"
-        output += self.start_time.strftime(format).lstrip("0").lower()
-        output += self.end_time.strftime("%I:%M%p").lower().lstrip("0")
-        return output
+        return '{}-{}'.format(self._human_readable_start_time(),
+                              self._human_readable_end_time())
+
+    def _human_readable_start_time(self):
```

Comment:
ya, `human_readable_time`, `_human_readable_start_time`, `_human_readable_end_time` & `human_readable_datetime`.

It's probably fine to just do `human_readable_datetime`.